### PR TITLE
Migrate `atomic-lock-free` to `rmake`

### PR DESCRIPTION
**Pull request by rejyr**
Also adds `llvm_components_contain` to `run-make-support`.

Part of #121876.

r? @jieyouxu

try-job: dist-x86_64-linux

### CODE REVIEW

File: tests/run-make/atomic-lock-free/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        .assert_stdout_not_contains("__atomic_fetch_add");
+}
+
+fn compile_and_check(target: &str) {
+    compile(target);
+    check();
+}
+
+fn main() {
+    if is_linux() {
```
Comment (fbstj) [inline tests/run-make/atomic-lock-free/rmake.rs:24]: does rmake support the `//@ only-linux` pragmas?

Comment (jieyouxu) [reply tests/run-make/atomic-lock-free/rmake.rs:24]: Yes, it should

Comment (jieyouxu) [reply tests/run-make/atomic-lock-free/rmake.rs:24]: Suggestion: Yeah please use `//@ only-linux` directive over the `is_linux` check here so the test is properly registered as "ignored" when running on non-linux instead of silently passing.

File: src/tools/run-make-support/src/lib.rs [hunk 2]
```diff
@@ -74,6 +74,12 @@ pub fn ar(inputs: &[impl AsRef<Path>], output_path: impl AsRef<Path>) {
     }
 }
 
+/// Check if target is linux-like.
+#[must_use]
+pub fn is_linux() -> bool {
+    target().contains("linux")
+}
```
Comment (jieyouxu) [inline src/tools/run-make-support/src/lib.rs:81]: Suggestion: and since we would like to prefer to use `//@ ignore-linux`, could you please revert this change.

**Review by jieyouxu — COMMENTED** [review]
Thanks! Just one suggestion for `//@ only-linux`. I'll also run some try jobs to check.

**Comment by jieyouxu** [comment]
Thanks!
@bors r+ rollup
